src/jit/inlinepolicy.h

```diff
@@ -155,6 +155,25 @@ class LegacyPolicy : public LegalPolicy
     bool                    m_ConstantFeedsConstantTest :1;
 };
 
+// NoThrowInlineLegacyPolicy extends the legacy policy by rejecting
+// inlining of methods that never return because they throw.
+
+class NoReturnLegacyPolicy : public LegacyPolicy
```

Comment:
Let's call this the `EnhancedLegacyPolicy` or something similar since there are other changes we might put in here down the road (eg handling STARG, pinned locals, etc...)